### ISSUE TITLE
simulations with 3 pops, 2 individuals

### ISSUE THREAD

**Issue by grenaud** [russcd/Ancestry_HMM#21]
Hello! we are trying to run ancestry HMM with simulated data for a single chromosome. Here YRI=Yoruba, CEU = central European and CHB = Han Chinese in Beijing.

We have a YRI+YRI individual and another who is YRI+YRI for the first 100Mb and then CEU+CEU.

We ran:
`nice -19 {input.ahmm_exe} -i {input.ahmm_input} -s ahmm.ploidy -a 3 0.25 0.75 0.0 -p 0 1 -0.25 -p 1 1 -0.75 -p 2 1 -0.01`

Our input looks like this:
```
1	144	1000	0	988	12	1000	0	0.01000144	2	0	2	0
1	163	6	994	179	821	3	997	1.9e-07	0	2	1	1
1	186	1000	0	1000	0	999	1	2.3e-07	2	0	2	0
1	188	994	6	821	179	997	3	2e-08	2	0	1	1
1	192	1000	0	1000	0	998	2	4e-08	2	0	2	0
1	227	6	994	179	821	3	997	3.5e-07	0	2	1	1
1	330	1000	0	1000	0	998	2	1.03e-06	2	0	2	0
1	336	976	24	606	394	915	85	6.000000000000001e-08	1	1	2	0
1	431	1000	0	1000	0	1000	0	9.5e-07	2	0	2	0
1	622	1000	0	904	96	1000	0	1.91e-06	2	0	2	0

```
where it is CEU, YRI and CHB.
And the output for individual 1 is:

```
chrom   position        2,0,0   1,1,0   1,0,1   0,2,0   0,1,1   0,0,2
1       144     0       0       0       0       0       1
1       163     0       0       0       0       0       1
1       186     0       0       0       0       0       1
1       188     0       0       0       0       0       1
1       192     0       0       0       0       0       1

```

Then we have a bunch of -nan.

However, it should be 100% 0,2,0. We ran f2 stats to check (CEU, YRI and CHB in order):
```
head -n 1000  ahmm_input_file_vcf.panel| awk '{totalPop1=$3+$4; freqA_pop1=$3/totalPop1; freqB_pop1=$4/totalPop1; totalPop2=$10+$11; freqA_pop2=$10/totalPop2; freqB_pop2=$11/totalPop2; F2=(freqA_pop1-freqA_pop2)^2+(freqB_pop1-freqB_pop2)^2; print F2;}' | awk 'BEGIN {sum=0} {sum += $1} END {print "number "NR" sum "sum" avg "sum/NR}'
number 1000 sum 49.905 avg 0.049905

head -n 1000  ahmm_input_file_vcf.panel| awk '{totalPop1=$5+$6; freqA_pop1=$5/totalPop1; freqB_pop1=$6/totalPop1; totalPop2=$10+$11; freqA_pop2=$10/totalPop2; freqB_pop2=$11/totalPop2; F2=(freqA_pop1-freqA_pop2)^2+(freqB_pop1-freqB_pop2)^2; print F2;}' | awk 'BEGIN {sum=0} {sum += $1} END {print "number "NR" sum "sum" avg "sum/NR}'
number 1000 sum 24.4245 avg 0.0244245

head -n 1000  ahmm_input_file_vcf.panel| awk '{totalPop1=$7+$8; freqA_pop1=$7/totalPop1; freqB_pop1=$8/totalPop1; totalPop2=$10+$11; freqA_pop2=$10/totalPop2; freqB_pop2=$11/totalPop2; F2=(freqA_pop1-freqA_pop2)^2+(freqB_pop1-freqB_pop2)^2; print F2;}' | awk 'BEGIN {sum=0} {sum += $1} END {print "number "NR" sum "sum" avg "sum/NR}'
number 1000 sum 49.2466 avg 0.0492466
```

So the lowest f2 is with YRI. Any pointers? Thank you ahead for any help!

**Comment by russcd** [comment]
Hi Gabriel, 

I think I have an idea what's going on. There's a couple of things:

1. I think the main issue is that you have a pulse that does not contribute any ancestry. I am not 100% sure, but I suspect that's the main problem. So, you could either remove that ancestry type or you might prefer to change the input to "-a 3 0.25 0.74 0.01" or something like that so that it contributes a very small amount to the ancestry fraction. 
2. A smaller concern is that AHMM will not estimate global ancestry proportions "-a 3 ...". So when you use negative values for each pulse, it's not estimating that parameter. 
3. It looks to me like you may be providing the program with genotypes rather than read counts. The reason I suspect this is that all individuals appear to have exactly 2 alleles observed. If so, please append "-g" on the command line.]

I suggest trying this:

`nice -19 {input.ahmm_exe} -i {input.ahmm_input} -s ahmm.ploidy -a 3 0.25 0.74 0.01 -p 0 10000 0.25 -p 1 -10 0.74 -p 2 1 0.01 -g`

This will fit a single the time of introgression for population 1 only with population 0 as resident and population 2 as fixed with fraction 0.01 and time 1 generation prior to the present. 

If that does run correctly, I'd be happy to try to debug. Would you be willing to send me an example input file?

**Comment by russcd** [comment]
The issue results from underflow due to relatively large reference panels. This can be resolved here by appending "--fix" on the command line which fixes teh ancestral allele frequencies. 

I'm going to close this for now and get to work on a bugfix.
